mautic/mautic
# html_entity_decode in the form submissions table can cause XSS - removed

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | N
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/3210
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
html_entity_decode was used in the form submissions table to display HTML formatting instead of the HTML markup. But this can cause XSS attack so I suggest to remove it.

#### Steps to test this PR:
1. Apply this PR and refresh the form results table.
2. The table should display both rows and the content you submitted with the HTML markup.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Create a new form with a textarea or a text field.
2. Open the form preview, fill in this value: `<script>document.write("busted");<\script>`
3. Submit the form with the same value again.
4. Check the form results. In my case I could see only one row instead of 2. In case of the linked issue, there was no table at all. In any case, the form results view should be broken.
